plugins/finale-paper/src/main/java/com/github/maxopoly/finale/combat/CombatUtil.java

```diff
@@ -73,8 +67,6 @@ public static void attack(ServerPlayer attacker, Entity victim) {
                 shouldDamage = f2 > 0.9f;
                 damage *= 0.2F + f2 * f2 * 0.8F;
                 f1 *= f2;
-            } else if (victim instanceof LivingEntity living) {
-                shouldDamage = living.invulnerableTime <= living.invulnerableDuration / 2;
```

Comment:
This is a line of code that took me several hours to write. Unfortunately, this is necessary because of the way that Minecraft handles knockback and damage. Removing this just introduces the bug where you can deal knockback with no cooldwon. I think that Paper should be fixing this in 1.21.3 so that may be a better way.